### DISCUSSION TITLE
Extension error: Could not import extension myst_parser

### DISCUSSION THREAD

**Discussion by LucyWriter** [executablebooks/MyST-Parser#573] [Q&A]
Greetings - has anyone experienced this error and fixed it?

Running Sphinx v4.5.0

Extension error:
Could not import extension myst_parser (exception: No module named 'myst_parser')
make: *** [html] Error 2

**Comment by chrisjsewell** [comment]
Heya, it appears that you have not installed myst-parser

**Reply by LucyWriter** [reply]
But I see this when I install - pip3 install myst-parser
Requirement already satisfied: myst-parser in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (0.17.2)
Collecting sphinx<5,>=3.1
  Using cached Sphinx-4.5.0-py3-none-any.whl (3.1 MB)

**Reply by chrisjsewell** [reply]
But perhaps you are running sphinx-build installed from a different Python environment?

**Reply by LucyWriter** [reply]
How can I check that?

**Reply by chrisjsewell** [reply]
Maybe `which sphinx-build`, can't remember if you can run `pip -m sphinx` 🤔

**Reply by LucyWriter** [reply]
Nothing seems to work - I've installed, uninstalled sphinx-doc and the extensions. 
Running Sphinx v4.5.0

Extension error:
Could not import extension myst_parser (exception: No module named 'myst_parser')
make: *** [html] Error 2

Using this command worked before - pip3 install git+https://github.com/sphinx-doc/sphinx but now it's pulling in 5.1.0+ and I see error messages about incompatibility. Uninstalled it and then used brew install sphinx-doc. Still broken.

**Comment by Daraan** [comment]
I also had this problem. It was solved after I activated my environment once more after the installation.

Thank you past self; as I again ran into this problem.

**Comment by 72757373656c6c** [comment]
I know this discussion is a little old, but perhaps this might help someone in the future. Regarding the original issue. As Daraan pointed out, this is a $PATH problem. The directory where myst-parser was installed was not in the $PATH.  

When I began in the sphinx+myst world, I followed [Paul Everitt's steps](https://www.jetbrains.com/guide/python/tutorials/sphinx_sites/setup/), which creates a python virtual environment. 

To summarize with some modifications (YMMV :smile:):

1. Create your documentation directory.

       mkdir -p src/project && cd src/project

2. Install python virtual environment 

       python -m venv .venv

3. Activate virtual environment. Notice the terminal prompt change and the $PATH change.

       source .venv/bin/activate

4. Create the file, `requirements.txt` and add required packages. Change as needed.  At a minimum, you need sphinx and myst-parser. The default theme is alabaster.

       sphinx
       myst-parser
       sphinx-book-theme
       sphinx-copybutton
       sphinx-togglebutton
       sphinx-design

5. Install requirements.

       pip install -r requirements.txt

6. Setup your sphinx documentation environment. There are a couple of prompts. See Paul's doc for a screenshot.

       sphinx-quickstart

7. Modify the conf.py as needed. Read the [theme's documentation](https://sphinx-book-theme.readthedocs.io/en/stable/index.html).

       cd source

       nano conf.py

       extensions = [
         "myst_parser",
         "sphinx_copybutton",
         "sphinx_design",
         "sphinx_togglebutton",
       ]

       myst_enable_extensions = [
         "attrs_block",
         "attrs_inline",
         "colon_fence",
         "deflist",
       ]

       source_suffix = { ".md": "markdown" }
       html_theme = "sphinx_book_theme"
       #html_theme = "alabaster"

8. Assuming all markdown, remove index.rst and add your own index.md file.

       nano index.md

       # Welcome 

       :::{toctree}
       :glob:
       :hidden:
       :maxdepth: 1

       [a-z]*
       :::

9. Create your documentation. Files end with `.md`. Read [myst-parser documentation](https://myst-parser.readthedocs.io/en/latest/).

10. Create your html documentation. See documentation for other [builders](https://www.sphinx-doc.org/en/master/usage/builders/index.html).

        cd ../
        make clean 
        make html
        firefox build/html/index.html

**Reply by agoose77** [reply]
Thanks for writing up the steps that you took here @72757373656c6c :) Python environment handling can be a source of confusion for new users, so it's good to have a comprehensive reply to point to :)
